sklearn/linear_model/quantile.py

```diff
@@ -0,0 +1,315 @@
+# Authors: David Dale dale.david@mail.ru
+# License: BSD 3 clause
+
+import numpy as np
+import warnings
+from scipy import optimize
+
+from ..base import BaseEstimator, RegressorMixin
+from .base import LinearModel
+from ..utils import check_X_y
+from ..utils import check_consistent_length
+from ..utils.extmath import safe_sparse_dot
+
+
+def _smooth_quantile_loss_and_gradient(
+        w, X, y, quantile, alpha, l1_ratio, sample_weight, gamma=0):
+    """ Smooth approximation to quantile regression loss, gradient and hessian.
+    Main loss and l1 penalty are both approximated by the same trick
+    from Chen & Wei, 2005
+    """
+    _, n_features = X.shape
+    fit_intercept = (n_features + 1 == w.shape[0])
```

Comment:
Better to simply pass `fit_intercept` as parameter from the parent function: I find this indirect determinations confusing.